### DISCUSSION TITLE
Possible bug with `expandedOne` in `onModelAfterCreate`

### DISCUSSION THREAD

**Discussion by benallfree** [pocketbase/pocketbase#4120] [Q&A] [answered]
Here is an example where `expandedOne` returns `null` but `findFirstRecordByData` will succeed in finding the record. I'm not sure if I'm making a mistake but this seems like it could be a bug.

```js
onModelAfterCreate((e) => {
  const notificationRec = /** @type {models.Record} */ (e.model)

  // This works
  const foo = $app
    .dao()
    .findFirstRecordByData(
      `message_templates`,
      `id`,
      notificationRec.getString(`message_template`),
    )
  console.log(JSON.stringify(foo))

  // This does not work
  const messageTemplateRec = notificationRec.expandedOne('message_template')
  console.log(JSON.stringify(messageTemplateRec))
}, `notifications`)
```

**Comment by ganigeorgiev** [comment, marked as answer]
It is not a bug. The `expandedOne` helper just provide access to the expanded data but in the above you don't seem to have any expand.

For expanding records programmatically, please check the example in https://pocketbase.io/docs/js-records/#programmatically-expanding-relations.

**Reply by benallfree** [reply]
Doh!
